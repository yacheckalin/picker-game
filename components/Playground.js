import React, { useState } from "react";
import GridMessage from "./GridMessage";
import GridIntro from "./GridIntro";
import GridContainer from "./GridContainer";

import PropTypes from "prop-types";

import { DEFAULT_FIELD_SIZE, DEFAUTL_FIELD_HASH } from "../constants";

import { LEVELS } from "../levels";

const Playground = ({ menuBar }) => {
  const [logMessage, setLogMessage] = useState(
    "You can make a move! [right, left, up, down, space, alt, enter]"
  );
  const [grid, setGrid] = useState(LEVELS[0][1]); // use default level
  const [gridSize, setGridSize] = useState(DEFAULT_FIELD_SIZE);
  const [gridHash, setGridHash] = useState(DEFAUTL_FIELD_HASH);

  const handleLoadLevel = ([gridHash, grid, gridSize]) => {
    setGrid(grid);
    setGridHash(gridHash);
    setGridSize(gridSize);
  };

  return (
    <div className="row">
      {menuBar && <GridIntro loadHandler={handleLoadLevel} />}
      <GridContainer
        handleMessage={setLogMessage}
        mapper={grid}
        size={gridSize}
        key={gridHash}
      />
      <GridMessage log={logMessage} />
    </div>
  );
};

Playground.propTypes = {
  menuBar: PropTypes.bool.isRequired,
};

Playground.defaultProps = {
  menuBar: true,
};

export default Playground;
