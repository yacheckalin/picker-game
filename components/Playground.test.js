import React from "react";
import Playground from "./Playground";
import toJSON from "enzyme-to-json";
import { shallow } from "enzyme";

describe("<Playground />", () => {
  const props = {};

  it("render and displays properly", () => {
    const wrapper = shallow(<Playground />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
