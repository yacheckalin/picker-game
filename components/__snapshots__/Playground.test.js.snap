// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Playground /> render and displays properly 1`] = `
<div
  className="row"
>
  <GridIntro
    loadHandler={[Function]}
  />
  <GridContainer
    handleMessage={[Function]}
    key="level_default"
    mapper={
      Array [
        Array [
          8,
          0,
          1,
          0,
        ],
        Array [
          0,
          1,
          0,
          Array [
            90,
            "BLUE",
          ],
        ],
        Array [
          0,
          1,
          0,
          0,
        ],
        Array [
          0,
          0,
          0,
          Array [
            20,
            "BLUE",
          ],
        ],
      ]
    }
    size={4}
  />
  <Memo()
    log="You can make a move! [right, left, up, down, space, alt, enter]"
  />
</div>
`;
